articles/js-and-web-apis/index.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/js-and-web-apis/
+title: How Javascript Uses Web Browser Apis to Achieve Asynchronisity
+description: 
+author: adrian-murage
+date: 
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/js-and-web-apis/hero.jpg
+    alt: JS and Web APIs
+---
+### Introduction
+Pure JavaScript has a single thread of execution. Known as the [**main thread**](https://developer.mozilla.org/en-US/docs/Glossary/main_thread) in the browser context. "This means that long-running JavaScript functions can block the thread, leading to an unresponsive page and a bad user experience."[(MDN docs)](https://developer.mozilla.org/en-US/docs/Glossary/main_thread)
+
+Examples of long-running JavaScript functions are those that:
+- Fetch data from an API.
+- Wait for some time to elapse before executing.
+
+But, the above-mentioned functionality is commonly used in today's web applications. In this guide, you will learn how this functionality is achieved Asynchronously with Web Browser APIs.
+
+With an understanding of Asynchronous functionality you should be able to:
+1. Run long-running JavaScript functions without blocking the Thread Of Execution.
+2. Run functionality related to long-running JavaScript functions once they are completed.
+3. Understand what happens during the entire process.
+
+### Prerequisites
+Before you begin this guide you will need:
+- An understanding of what happens when [JavaScript executes your code](/engineering-education/js-execution-under-the-hood/).
+
+### Step 1 - When do you need Asynchronous JavaScript?
+```JavaScript
+1. function display(data){
+2.   console.log(data)
+3.   } 
+4. const dataFromAPI = fetchAndWait('https://twitter.com/will/tweets/1')
```

Comment:
please note that it should be
`fetch('https://twitter.com/will/status/1')`

Reply:
@Manley12 I actually intentionally used `fetchAndWait` which is in this case a made-up function.
Since the fetch function returns a promise object and has behaviors that I did not desire to go into in this tutorial. And I pointed this out in the paragraph below this